src/Features/Core/Portable/EmbeddedLanguages/RegularExpressions/RegexEmbeddedCompletionProvider.cs

```diff
@@ -0,0 +1,560 @@
+            context.AddIfMissing($"(?>  {regex_subexpression}  )", regex_nonbacktracking_subexpression_short, regex_nonbacktracking_subexpression_long, parentOpt, positionOffset: "(?>".Length, insertionText: "(?>)");
+
+            context.AddIfMissing($"(?(  {regex_expression}  )  {regex_yes}  |  {regex_no}  )", regex_conditional_expression_match_short, regex_conditional_expression_match_long, parentOpt, positionOffset: "(?(".Length, insertionText: "(?()|)");
+            context.AddIfMissing($"(?(  {regex_name_or_number}  )  {regex_yes}  |  {regex_no}  )", regex_conditional_group_match_short, regex_conditional_group_match_long, parentOpt, positionOffset: "(?(".Length, insertionText: "(?()|)");
+
+            context.AddIfMissing($"(?#  {regex_comment}  )", regex_inline_comment_short, regex_inline_comment_long, parentOpt, positionOffset: "(?#".Length, insertionText: "(?#)");
+            context.AddIfMissing($"(?imnsx-imnsx)", regex_inline_options_short, regex_inline_options_long, parentOpt, positionOffset: "(?".Length, insertionText: "(?)");
+            context.AddIfMissing($"(?imnsx-imnsx:  {regex_subexpression}  )", regex_group_options_short, regex_group_options_long, parentOpt, positionOffset: "(?".Length, insertionText: "(?:)");
+        }
+
+        private void ProvideCharacterClassCompletions(EmbeddedCompletionContext context, RegexNode parentOpt)
+        {
+            context.AddIfMissing($"[  {regex_character_group}  ]", regex_positive_character_group_short, regex_positive_character_group_long, parentOpt, positionOffset: "[".Length, insertionText: "[]");
+            context.AddIfMissing($"[  firstCharacter-lastCharacter  ]", regex_positive_character_range_short, regex_positive_character_range_long, parentOpt, positionOffset: "[".Length, insertionText: "[-]");
+            context.AddIfMissing($"[^  {regex_character_group}  ]", regex_negative_character_group_short, regex_negative_character_group_long, parentOpt, positionOffset: "[^".Length, insertionText: "[^]");
+            context.AddIfMissing($"[^  firstCharacter-lastCharacter  ]", regex_negative_character_group_short, regex_negative_character_range_long, parentOpt, positionOffset: "[^".Length, insertionText: "[^-]");
+            context.AddIfMissing($"[  {regex_base_group}  -[  {regex_excluded_group}  ]", regex_character_class_subtraction_short, regex_character_class_subtraction_long, parentOpt, positionOffset: "[".Length, insertionText: "[-[]]");
+        }
+
+        private void ProvideEscapeCategoryCompletions(EmbeddedCompletionContext context)
+        {
+            foreach (var (name, (shortDesc, longDesc)) in RegexCharClass.EscapeCategories)
+            {
+                var displayText = name;
+                if (displayText.StartsWith("_"))
+                {
+                    continue;
```

Comment:
So we have a bunch of entries in the list we just unconditionally throw out? Deserves a comment at least!

Reply:
sure.